src/main/java/hu/oe/nik/szfmv/automatedcar/visualization/debug/DebugViewer.java

```diff
@@ -0,0 +1,72 @@
+package hu.oe.nik.szfmv.automatedcar.visualization.debug;
+
+import hu.oe.nik.szfmv.automatedcar.visualization.Utils.DrawingInfo;
+
+import java.awt.*;
+
+
+/**
+ * Displays debugging-related information on the screen
+ */
+public class DebugViewer {
+
+    private Graphics2D graphics2D;
+    private DrawingInfo info = new DrawingInfo(Color.GREEN, 2);
+
+    /**
+     * @return The information represening the color and borderline width of the object
+     */
+    public DrawingInfo getInfo() {
+        return info;
+    }
+
+    /**
+     * @param info The information represening the color and borderline width of the object
+     */
+    public void setInfo(DrawingInfo info) {
+        this.info = info;
+    }
+
+    private boolean debuggerSwitchedOn = true;
+
+    /**
+     * @return If debugger is switched on or not
+     */
+    public boolean isdebuggerSwitchedOn() {
+        return this.debuggerSwitchedOn;
+    }
+
+    /**
+     * @param debuggerSwitchedOn Flag indicating if debugger is switched on or not
+     */
+    public void setdebuggerSwitchedOn(boolean debuggerSwitchedOn) {
```

Comment:
rename to: setDebuggerSwitchedOn
make it work like an actual switch, there is no need for a parameter, because if it's on the only other state it can take is off

Reply:
I intentionally used this version, because with switch-like implementations, you always need a get call before a set so that you know what the actual state is. This way, you just set the property to whatever value you want and you don't need o worry about its previous state.

Reply:
partially resolved (bool param still there) : c9ebfbbb773d17e5cbd7e2126d91314275e7bb88
I would ask for a 2nd opinion before changing this: @zeatrun,

Reply:
If you keep track of the usage, this shouldn't be a problem, especially with a debug feature. But you also made a point, so you get the approve